pelican/settings.py

```diff
@@ -156,79 +162,67 @@ def load_source(name, path):
 
 
 def read_settings(path=None, override=None):
+    settings = override or {}
+
     if path:
-        local_settings = get_settings_from_file(path)
-        # Make the paths relative to the settings file
+        settings = dict(get_settings_from_file(path), **settings)
+
+    if settings:
+        settings = handle_deprecated_settings(settings)
+
+    if path:
+        # Make relative paths absolute
+        def getabs(path, pluginpath):
```

Comment:
`pluginpath` not the best variable name? 

Probably,

```python
def getabs(base_path, path):
```

or since `base_path` is always `path`, maybe:
```python
def getabs(path, base_path=path):
```

Reply:
Agree on the name, I am thinking `maybe_relative` for `pluginpath`. I don't understand the rest of your comment: `path` is the path of the settings file, `pluginpath` is a relative path in the settings. (This function was already in the code, I just moved it and broadened its use.)

Reply:
`getabs` is used only in two places and always with `path` as the initial argument. So, I'm suggesting:
```python
        def getabs(path, settings_path=path):
            if isabs(path):
                return path
            else:
                path_dirname = os.path.dirname(settings_path)
                path_joined = os.path.join(path_dirname, path)
                path_normed = os.path.normpath(path_joined)
                path_absolute = os.path.abspath(path_normed)
                return path_absolute
```
and use it like:
```python
getabs(settings[p])
```

FWIW, second part could be:
```python
                return os.path.abspath(
                           os.path.normpath(
                           os.path.join(
                               os.path.dirname(settings_path), 
                               path)))
```
but I don't mind the more verbose one.